terraform/environments/core-shared-services/vpc.tf

```diff
@@ -26,4 +37,87 @@ module "vpc" {
   # Tags
   tags_common = local.tags
   tags_prefix = each.key
-}
\ No newline at end of file
+}
+
+locals {
+  # Derive non_live_data vpc and private subnets for configuration setting up the vpc endpoints
+  non_live_data_vpc_id = one([for k in module.vpc : k.vpc_id if k.vpc_cidr_block == local.networking.non_live_data])
+  private_subnet_ids   = flatten([for k in module.vpc : k.non_tgw_subnet_ids_map.private if k.vpc_cidr_block == local.networking.non_live_data])
+
+  # Get the private route table keys and values, then transform the values (the actual route table ids) into a list
```

Comment:
👏  thanks for the comment on the dataset transformation, these are always a nightmare to look back at 😄